cache-config/tm-health-client/tmutil/tmutil.go

```diff
@@ -0,0 +1,674 @@
+		caches, err := c.GetCacheStatuses()
+		if err != nil {
+			log.Errorln(err.Error())
+		}
+
+		for k, v := range caches {
+			hostName := string(k)
+			cs, ok := c.Parents[hostName]
+			if ok {
+				tmAvailable := *v.CombinedAvailable
+				if cs.available() != tmAvailable {
+					if !c.Cfg.EnableActiveMarkdowns {
+						if !tmAvailable {
+							log.Infof("TM reports that %s is not available and should be marked DOWN but, mark downs are disabled by configuration", hostName)
+						} else {
+							log.Infof("TM reports that %s is available and should be marked UP but, mark up is disabled by configuration", hostName)
+						}
+					} else {
+						if err = c.markParent(cs.Fqdn, *v.Status, tmAvailable); err != nil {
+							log.Errorln(err.Error())
+						}
+					}
+				}
+			}
+		}
+
+		time.Sleep(pollingInterval)
+	}
+}
+
+// Used by the polling function to update the parents list from
+// changes to parent.config and strategies.yaml.  The parents
+// availability is also updated to reflect the current state from
+// the trafficserver HostStatus subsystem.
+func (c *ParentInfo) UpdateParentInfo() error {
+	ptime, err := getFileModificationTime(c.ParentDotConfig.Filename)
+	if err != nil {
+		return errors.New("error reading parent.config: " + err.Error())
+	}
+	stime, err := getFileModificationTime(c.StrategiesDotYaml.Filename)
+	if err != nil {
+		return errors.New("error reading strategies.yaml: " + err.Error())
+	}
+	if c.ParentDotConfig.LastModifyTime < ptime {
+		// read parent.config
+		if err := c.readParentConfig(c.Parents); err != nil {
+			return errors.New("updating parent.config file: " + err.Error())
+		} else {
+			log.Infof("updated parents from new parent.config, total parents: %d\n", len(c.Parents))
+		}
+	}
+
+	if c.StrategiesDotYaml.LastModifyTime < stime {
+		// read strategies.yaml
+		if err := c.readStrategies(c.Parents); err != nil {
+			return errors.New("updating parent strategies.yaml file: " + err.Error())
+		} else {
+			log.Infof("updated parents from new strategies.yaml, total parents: %d\n", len(c.Parents))
+		}
+	}
+
+	// collect the trafficserver current host status
+	if err := c.readHostStatus(c.Parents); err != nil {
+		return errors.New("reading latest trafficserver host status: " + err.Error())
+	}
+
+	return nil
+}
+
+// choose an availabe trafficmonitor, returns an error if
+// there are none
+func (c *ParentInfo) findATrafficMonitor() (string, error) {
+	var tmHostname string
+	lth := len(c.Cfg.TrafficMonitors)
+	if lth == 0 {
+		return "", errors.New("there are no available traffic monitors")
+	}
+
+	// choose one at random
+	rand.Seed(time.Now().UnixNano())
+	r := (rand.Intn(7919) % lth)
+
+	tms := make([]string, 0)
+	for k, v := range c.Cfg.TrafficMonitors {
+		if v == true {
+			log.Debugf("traffic monitor %s is available\n", k)
+			tms = append(tms, k)
+		}
+	}
+
+	lth = len(tms)
+	if lth > 0 {
+		if r < lth { // return the trafficmonitor at r, r could be the first
```

Comment:
Nitpick: this `if` could be removed if `r` above was changed to `r := rand.Intn(lth)`

Reply:
Though it'll need to be moved inside `if lth > 0 { r := rand.Intn(lth)` because `rand.Intn(0)` panics

Reply:
yeah, I was playing with a primes, anyway I've changed it.